clients/android/app/src/main/java/app/lockbook/loggedin/mainscreen/MainScreenViewModelFactory.kt

```diff
@@ -0,0 +1,15 @@
+package app.lockbook.loggedin.mainscreen
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class MainScreenViewModelFactory(
+    private val path: String
+) : ViewModelProvider.Factory {
+    @Suppress("unchecked_cast")
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(MainScreenViewModel::class.java))
+            return MainScreenViewModel(path) as T
+        throw IllegalArgumentException("Unknown ViewModel class")
```

Comment:
What's happening here?

Reply:
This is boilerplate code that is needed to create an class extended by ViewModel. I use this in MainScreenActivity to create the viewmodel